src/Filter/Model/FilterData.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Filter\Model;
+
+/**
+ * @psalm-immutable
+ */
+final class FilterData
+{
+    /**
+     * @var ?int
+     */
+    private $type;
+
+    /**
+     * @var mixed
+     */
+    private $value;
+
+    /**
+     * @var bool
+     */
+    private $hasValue;
+
+    private function __construct()
+    {
+        $this->hasValue = false;
+    }
+
+    /**
+     * @psalm-pure
+     */
+    public static function fromArray(array $data): self
+    {
+        $filterData = new self();
+
+        if (isset($data['type'])) {
+            $type = $data['type'];
+
+            if (!is_numeric($type)) {
+                throw new \InvalidArgumentException(sprintf(
+                    'The "type" value MUST be "null" or "int", %s given.',
+                    \is_object($type) ? 'instance of '.\get_class($type) : \gettype($type)
+                ));
+            }
+
+            $filterData->type = (int) $type;
+        }
+
+        if (\array_key_exists('value', $data)) {
+            $filterData->value = $data['value'];
+            $filterData->hasValue = true;
+        }
+
+        return $filterData;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function value()
```

Comment:
Just to keep the consistency with other classes or models, I think we should use the "get" prefix for this getter.
```suggestion
    public function getValue()
```

Reply:
I think if we start to introduce VOs maybe would be nice to starting using this kind of naming (used also in the refactoring of @OskarStark in https://github.com/sonata-project/dev-kit/blob/master/src/Domain/).

But I agree about consistency, it's probably true that can be seen as strange to mix them.